### ISSUE TITLE
Updated reDIP-SID implementation?

### ISSUE THREAD

**Issue by Jammmmm** [MiSTer-devel/C64_MiSTer#179]
The SID implementation in C64 MiSTer seems to be from 2022. reDIP-SID has had many commits since then.

I'm just wondering if there's any plans to update the SID code to the latest reDIP-SID code?

**Comment by sorgelig** [comment]
Year has nothing to do with correctness of implementation. MiSTer implementation is not based on reDIP-SID. 
It's almost perfect and doesn't need changes.

**Comment by Jammmmm** [comment]
Oh, I thought it was based on reDIP-SID because files such as rtl/sid/sid_dac.sv say "This file is part of reDIP SID" and rtl/sid/sid_filter.sv says "This file is based on filter from reDIP-SID". My mistake!

I agree, it's definitely close to perfect but there are times when it doesn't work. I reported issue #167, but just a few weeks ago there was another game I encountered that had similar SID issues. Unfortunately I don't remember what game :(

**Comment by sorgelig** [comment]
SID implementation went through many changes. In latest version i've used some parts of reDIP-SID but not the whole implementation. 
Some games are targeted for specific SID revision. Did you try both 6581 and 8560 options? Did you try to enable/disable digi-fix in 8580 mode?

**Comment by Jammmmm** [comment]
ah! I haven't looked through the code or commit history so I just assumed it was largely reDIP-SID based.

I have tried both 6581 and 8580, and have tried turning the digi-fix on and off. Still the same problem.

Here is a CRT of the game that allows you to test it: [ThunderBolt.zip](https://github.com/user-attachments/files/17351868/ThunderBolt.zip). Start the game and you won't hear the plane flying audio on MiSTer.

It works fine in Vice with 6581 ReSID. It does not work at all with Fast SID or 8580.

**Comment by sorgelig** [comment]
I've tried to find a technical info how this game output samples, but couldn't find yet.

**Comment by Jammmmm** [comment]
Yeah. Unfortunately it's an obscure budget game that nobody talks about. I think the only way would be to disassemble it, which is time consuming.

Interestingly, the problem was also mentioned on [Lemon 64](https://www.lemon64.com/forum/viewtopic.php?t=85141) :)
